app/controllers/books_controller.rb

```diff
@@ -1,17 +1,38 @@
 class BooksController < ApplicationController
   before_action :redirect_to_admin_books, only: [:index]
-  before_action :redirect_to_sign_in, only: %i[index show], unless: :user_signed_in?
-  before_action :redirect_to_reservation_lending_show, only:[:show]
+  before_action :logged_in_user, only: %i[show]
```

Comment:
```suggestion
  before_action :redirect_to_sign_in, only: %i[show], unless: :user_signed_in?
```
application_controllerで `redirect_to_sign_in` を定義してるので確認してみてください 🙏